[pkg/ottl] Add a new ottl function Keys (#39256)
open-telemetry/opentelemetry-collector-contrib: pkg/ottl/ottlfuncs/README.md

- [UUID](#UUID)
- [Weekday](#weekday)
- [Year](#year)

> - [Keys](#keys)

Author: Right now it is not sorted, shall I sort all the lines?

Examples:

- `Year(Now())`

### Keys

> `Keys(target)`

Reviewer: ```suggestion
`Keys(target)`

```

> The `Keys` Converter returns a slice of keys from the given map.

Reviewer: ```suggestion
The `Keys` Converter returns a slice containing all the keys from the given map.

```

> `target` is a `pcommon.Map`. If `target` is not a `pcommon.Map`, an error will be returned.

Reviewer: ```suggestion
`target` is a `pcommon.Map`. If `target` is another type an error is returned.

```

The returned type is `pcommon.Slice`.

Examples:
- 

> - `Keys(resource.attributes)`

Reviewer: ```suggestion
- `Keys(resource.attributes)`
- `Keys({"k1":"v1", "k2": "v2"})`
```
